### PR TITLE
Error Handling

### PR DESCRIPTION
**Pull request by medhapatil12**
- Made changes to handle error scenarios.
- Added a service.tsx file to separate service call. 

https://github.com/Simhanischal/Mini-ChatGPT/assets/43102381/9e97b27d-38fe-4428-b35d-5fce15a416ce

### CODE REVIEW

File: src/services.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      });
+      newMessages.push(newMessage)
+      setMessages(newMessages);
+      setTypingIndicator(false);
+      setStatus('success');
+    })
+    .catch((error) => {
+      setTypingIndicator(false);
+      setStatus('failed');
+      console.log(error);
```
Comment (Simhanischal) [inline src/services.tsx:48]: nit: remove console log

File: src/components/chat/chatInput/ChatInput.tsx [hunk 2]
```diff
@@ -33,7 +34,7 @@ const ChatInput = (props: ChatInputProps) => {
         onChange={handleInputChange}
         onKeyDown={handleEnter}
       />
-      {input.length > 0 && (
+      {input.length > 0 && (status === '' || status === 'success') && (
```
Comment (Simhanischal) [inline src/components/chat/chatInput/ChatInput.tsx:37]: can be simplified by checking for `status !== 'failed'`?

**Review by Simhanischal — APPROVED** [review]
This was one of our most challenging features and you worked very hard to achieve it! I am loving the result! Thank you :)

**Comment by medhapatil12** [comment]
> This was one of our most challenging features and you worked very hard to achieve it! I am loving the result! Thank you :)

Thank you for the support! 🥺

**Pull request merged**